internetarchive/iaux-clearable-text-input
# WEBDEV-5399 Clearable text input (initial component)

- Renders a text input field that shows a clear button whenever the field contains text
- Hitting the clear button empties the text field and emits a custom 'clear' event, whose detail property is the value of the input field prior to clearing
- Has properties to set the field's value, placeholder text, and screenreader labels/controls
- Includes a number of custom CSS vars to aid with styling in different contexts

<img src="https://github.com/internetarchive/iaux-clearable-text-input/raw/webdev5399-clearable-text-input/component-example.gif" width="300">

### Testing:
- Open the demo app: https://internetarchive.github.io/iaux-clearable-text-input/pr/pr-1/demo/
- The input field should be blank (with placeholder text shown) and no clear button
- Typing in the field should immediately show the clear button
- Manually erasing what you typed should make the clear button disappear
- Typing in the field and then pressing the clear button should erase the contents of the input field
- Reload the demo app with a `?query=something` parameter in the URL: https://internetarchive.github.io/iaux-clearable-text-input/pr/pr-1/demo/?query=something
- The field should show your query as its initial value and the clear button should be initially visible
- As before, hitting the clear button should erase the input field's contents
